perses/dispersed/feptasks.py

```diff
@@ -291,222 +158,210 @@ def __init__(self, alchemical_functions: dict, splitting: str, temperature: unit
-        ----------
-        thermodynamic_state : openmmtools.states.ThermodynamicState
-            thermodynamic state; unused here.
+        self._timers['instantiate'] = time.time() - start
 
-        Returns
-        -------
-        integrator : openmmtools.integrators.AlchemicalNonequilibriumLangevinIntegrator
-            The integrator that is associated with this MCMove
-        """
-        return self._integrator
+        #set a bool variable for pass or failure
+        self.succeed = True
 
-    def apply(self, thermodynamic_state, sampler_state):
+    def apply(self, thermodynamic_state, sampler_state, lambda_protocol = 'default'):
         """Propagate the state through the integrator.
         This updates the SamplerState after the integration. It will apply the full NCMC protocol.
-
         Parameters
         ----------
-        thermodynamic_state : openmmtools.states.ThermodynamicState
-           The thermodynamic state to use to propagate dynamics.
+        thermodynamic_state : openmmtools.states.CompoundThermodynamicState
+           The compound thermodynamic state to use to propagate dynamics.
         sampler_state : openmmtools.states.SamplerState
            The sampler state to apply the move to. This is modified.
+        lambda_protocol : str (or dict), default 'default'
+           Which protocol to use for the lambda update
         """
         """Propagate the state through the integrator.
         This updates the SamplerState after the integration. It also logs
         benchmarking information through the utils.Timer class.
-        Parameters
-        ----------
-        thermodynamic_state : openmmtools.states.ThermodynamicState
-           The thermodynamic state to use to propagate dynamics.
-        sampler_state : openmmtools.states.SamplerState
-           The sampler state to apply the move to. This is modified.
         See Also
         --------
         openmmtools.utils.Timer
         """
+        start = time.time()
+        thermodynamic_state = copy.deepcopy(thermodynamic_state)
+        sampler_state = copy.deepcopy(sampler_state)
+
         # Check if we have to use the global cache.
         if self.context_cache is None:
             context_cache = cache.global_context_cache
         else:
             context_cache = self.context_cache
 
-        # Create integrator.
-        integrator = self._get_integrator(thermodynamic_state)
+        #define the lambda schedule (linear)
+        if self._direction == 'forward':
+            start_lambda = 0.0
+            end_lambda = 1.0
+        elif self._direction == 'reverse':
+            start_lambda = 1.0
+            end_lambda = 0.0
+        else:
+            raise Error(f"direction must be 'forward' or 'reverse'")
 
-        context, integrator = context_cache.get_context(thermodynamic_state, integrator)
+        # define the lambda_schedule
+        lambda_schedule = np.linspace(start_lambda, end_lambda, self._nsteps + 1)[1:]
```

Comment:
Not for now, but this might be something we would want to move into the `LambdaProtocol` object at some point.

out of interest... I guess there is no reason that this has to be linear?

Reply:
So the `LambdaProtocol` would contain both the functions by which the energy terms change, and the schedule by which things are changing too, which could also contain a lot of functions to 'validate' the choice in protocol.

Reply:
in the future, when we explore nonlinear lambda interpolations, I'll make a small PR to expose this as an option